c7n/filters/core.py

```diff
@@ -1018,3 +1020,123 @@ def reorder(self, items, key=None):
             return items[::-1]
         else:
             return sorted(items, key=key, reverse=(self.order == 'desc'))
+
+
+class FakeModel:
+    id = 'c7n:_id'
+
+
+class FakeFilterRegistry(FilterRegistry):
+
+    def __init__(self, *args, **kw):
+        super(FilterRegistry, self).__init__(*args, **kw)
+        self.register('value', ValueFilter)
+        self.register('or', Or)
+        self.register('and', And)
+        self.register('not', Not)
+        self.register('event', EventFilter)
+        self.register('reduce', ReduceFilter)
+
+
+class FakeResourceManager(ResourceManager):
+    filter_registry = FakeFilterRegistry('fake')
+
+    def get_model(self):
+        return FakeModel()
+
+
+class ListItemFilter(ValueFilter):
+    """
+    Filter on attributes on items on a list
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: find-task-def-not-using-registry
+            resource: aws.ecs-task-definition
+            filters:
+              - not:
+                - type: list-item
+                  key: containerDefinitions
+                  value:
+                    - not:
+                      - type: value
```

Comment:
its probably best to use a multi attribute, else the extant value filter could be used easily for the same use case, ie example should cover the unique value of this filter.